packages/components/src/RecurringSelect/RecurringSelect.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { MonthlySelect } from "./components";
+import { CurrentRecurrenceComponent } from "./CurrentRecurrenceComponent";
+import styles from "./RecurringSelect.css";
+import {
+  DurationPeriod,
+  RecurrenceRule,
+  isMonthly,
+  typeInitialStateMap,
+} from "./types";
+import { Option, Select } from "../Select";
+import { InputNumber } from "../InputNumber";
+import { InputGroup } from "../InputGroup";
+import { Content } from "../Content";
+import { Text } from "../Text";
+
+interface RecurringSelectProps {
+  disabled?: boolean;
+  recurrenceRule: RecurrenceRule;
```

Comment:
just for consistency with other components in the atlantis. 
```suggestion
  readonly disabled?: boolean;
  readonly recurrenceRule: RecurrenceRule;
```

also, this probably should be `value` since it's technically a huge input. Make it easy to guess what the prop be if most of our controlled components uses a `value` and `onChange` combo